pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -45,59 +51,88 @@ var log = logf.Log.WithName("nodeconfig")
 	if err := nc.configureNetwork(); err != nil {
 		return errors.Wrap(err, "configuring node network failed")
 	}
+	// Apply worker labels
+	if err := nc.applyWorkerLabel(); err != nil {
+		return errors.Wrap(err, "failed applying worker label")
+	}
 	log.Info("VM has been configured as a worker node", "VM ID", nc.GetCredentials().GetInstanceId())
 	return nil
 }
 
 // configureNetwork configures k8s networking in the node
 func (nc *nodeConfig) configureNetwork() error {
-	if nc.GetCredentials() == nil {
-		return fmt.Errorf("nil credentials for VM")
-	}
-
-	if nc.GetCredentials().GetIPAddress() == "" {
-		return fmt.Errorf("empty IP for VM: %v", nc.GetCredentials())
-	}
-
-	node, err := nc.getNode(nc.GetCredentials().GetIPAddress())
-	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("error getting node object for VM %s",
-			nc.GetCredentials().GetInstanceId()))
-	}
-
 	// Wait until the node object has the hybrid overlay subnet annotation. Otherwise the hybrid-overlay will fail to
 	// start
-	if err = nc.waitForNodeAnnotation(node.GetName(), HybridOverlaySubnet); err != nil {
+	if err := nc.waitForNodeAnnotation(nc.node.GetName(), HybridOverlaySubnet); err != nil {
 		return errors.Wrap(err, fmt.Sprintf("error waiting for %s node annotation for %s", HybridOverlaySubnet,
-			node.GetName()))
+			nc.node.GetName()))
 	}
 
 	// NOTE: Investigate if we need to introduce a interface wrt to the VM's networking configuration. This will
 	// become more clear with the outcome of https://issues.redhat.com/browse/WINC-343
 
 	// Configure the hybrid overlay in the Windows VM
-	if err = nc.Windows.ConfigureHybridOverlay(node.GetName()); err != nil {
-		return errors.Wrap(err, fmt.Sprintf("error configuring hybrid overlay for %s", node.GetName()))
+	if err := nc.Windows.ConfigureHybridOverlay(nc.node.GetName()); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error configuring hybrid overlay for %s", nc.node.GetName()))
 	}
 
 	// Wait until the node object has the hybrid overlay MAC annotation. This is required for the CNI configuration to
 	// start.
-	if err = nc.waitForNodeAnnotation(node.GetName(), HybridOverlayMac); err != nil {
+	if err := nc.waitForNodeAnnotation(nc.node.GetName(), HybridOverlayMac); err != nil {
 		return errors.Wrap(err, fmt.Sprintf("error waiting for %s node annotation for %s", HybridOverlayMac,
-			node.GetName()))
+			nc.node.GetName()))
+	}
+	return nil
+}
+
+// getInstanceID gets the instanceID of VM for a given cloud provider ID
+// Ex: aws:///us-east-1e/i-078285fdadccb2eaa. We always want the last entry which is the instanceID
+func getInstanceID(providerID string) string {
+	providerTokens := strings.Split(providerID, "/")
+	return providerTokens[len(providerTokens)-1]
+}
+
+// applyWorkerLabel applies the worker label to the Windows Node we created.
+func (nc *nodeConfig) applyWorkerLabel() error {
+	nc.node.Labels[WorkerLabel] = ""
```

Comment:
Just to be on the safe side, have a guard if the label is already present and continue..